### PR TITLE
Added Container List Context Menu actions and Added Service to handle them

### PR DESCRIPTION
**Pull request by motasimvd**
### Proposed Changes
* Added Container List Context Menu
* Added Service To Handle Context Menu Actions

### Checklist
- [x] Tests
- [x] Translations
- [ ] Security Implications Contemplated (add notes if applicable)

### Additional Info
** Changes are WIP, Merging to canary to use dotCMS as base **

### Screenshots
Original             |  Updated
:-------------------------:|:-------------------------:
** original screenshot **  |  ** updated screenshot **

image

### CODE REVIEW

**Review by rjvelazco — CHANGES_REQUESTED** [review]

File: core-web/apps/dotcms-ui/src/app/api/services/dot-containers/dot-containers.service.ts [hunk 1]
```diff
@@ -0,0 +1,187 @@
...
+            pluck('entity'),
+            catchError((error: HttpErrorResponse) => {
+                return this.httpErrorManagerService.handle(error).pipe(
+                    take(1),
+                    map(() => null)
+                );
+            })
+        );
+    }
+}
```
[the remark this answers is not in the document]
Comment (motasimvd) [reply core-web/apps/dotcms-ui/src/app/api/services/dot-containers/dot-containers.service.ts:187]: Trying not to break the Workaround due to type mapping, this was just copied from template Service

File: core-web/apps/dotcms-ui/src/app/api/services/dot-containers/dot-containers.service.ts [hunk 2]
```diff
@@ -0,0 +1,187 @@
...
+     * @param {DotContainer} values
+     * @return Observable<DotContainer>
+     * @memberof DotContainersService
+     */
+    create(values: DotContainer): Observable<DotContainer> {
+        return this.request<DotContainer>({ method: 'POST', url: CONTAINER_API_URL, body: values });
+    }
+
+    /**
+     * Updates a template
```
Comment (hmoreras) [inline core-web/apps/dotcms-ui/src/app/api/services/dot-containers/dot-containers.service.ts:78]: missing param in doc

File: core-web/apps/dotcms-ui/src/app/api/services/dot-containers/dot-containers.service.ts [hunk 3]
```diff
@@ -0,0 +1,187 @@
...
+     * @memberof DotContainersService
+     */
+    archive(identifiers: string[]): Observable<DotActionBulkResult> {
+        const url = `${CONTAINER_API_URL}_archive`;
+
+        return this.request<DotActionBulkResult>({ method: 'PUT', url, body: identifiers });
+    }
+
+    /**
+     * Unpublish a template00
```
Comment (hmoreras) [inline core-web/apps/dotcms-ui/src/app/api/services/dot-containers/dot-containers.service.ts:139]: 00?

File: core-web/apps/dotcms-ui/src/app/portlets/dot-containers/container-list/container-list.component.ts [hunk 4]
```diff
@@ -1,8 +1,70 @@
...
 
 @Component({
     selector: 'dot-container-list',
     templateUrl: './container-list.component.html',
-    styleUrls: ['./container-list.component.scss']
+    styleUrls: ['./container-list.component.scss'],
+    providers: [DotContainerListStore]
 })
-export class ContainerListComponent {}
+export class ContainerListComponent {
```
Comment (alfredo-dotcms) [inline core-web/apps/dotcms-ui/src/app/portlets/dot-containers/container-list/container-list.component.ts:14]: this files and all the files in the **container-list** directory needs to have the prefix **"dot-"**

ex: DotContainerListComponent

File: core-web/apps/dotcms-ui/src/app/portlets/dot-containers/container-list/container-list.component.ts [hunk 5]
```diff
@@ -1,8 +1,70 @@
...
+    /**
+     * Reset bundle state to null
+     *
+     * @memberof ContainerListComponent
+     */
+    resetBundleIdentifier(): void {
+        this.store.updateBundleIdentifier(null);
+    }
+
+    setContainerActions(container: DotContainer): DotActionMenuItem[] {
```
Comment (alfredo-dotcms) [inline core-web/apps/dotcms-ui/src/app/portlets/dot-containers/container-list/container-list.component.ts:67]: doc?

File: core-web/apps/dotcms-ui/src/app/api/services/dot-router/dot-router.service.ts [hunk 6]
```diff
@@ -197,6 +197,20 @@ export class DotRouterService {
...
+     * Redirect to edit the template.
+     * If the inode is passed, load a specific version of the template
+     *
+     * @param {string} id
+     * @param {string} inode
+     * @memberof DotRouterService
+     */
+    goToEditContainer(id: string, inode?: string): void {
+        this.router.navigate([
+            inode ? `/container-new/edit/${id}/inode/${inode}` : `/container-new/edit/${id}`
```
Comment (fmontes) [inline core-web/apps/dotcms-ui/src/app/api/services/dot-router/dot-router.service.ts:210]: Let's replace the route to be `containers`.

Let's catch that as soon as possible.

**Review by fmontes — CHANGES_REQUESTED** [review]

**Pull request closed without merging**